sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -1077,6 +1092,55 @@ public Param param(String key) {
       return tags;
     }
 
+    /**
+     * Deprecated rules keys for this rule.
+     * <p>
+     * If you want to rename the key of a rule or change its repository or both, register the rule's previous repository
+     * and key.
+     * <br>
+     * This will allow SonarQube to support "issue renaming" (ie. <strong>not</strong> closing existing issues
+     * for this rules, created under the rule's previous repository and/or name, and recreate them as new under the
+     * new issue's repository and/or key).
+     * <p>
+     * Several deprecated keys can be provided to allow SonarQube to support "issue renaming" across multiple versions
+     * of a plugin.
+     * <br>
+     * Consider the following use case scenario:
+     * <ul>
+     *   <li>Rule {@code Foo:A} is defined in version 1 of the plugin</li>
+     *   <li>Rule's key is renamed to B in version 2 of the plugin
+     *      <ul><li>rule {@code Foo:B} defines {@code Foo:A} as a deprecated key</li></ul>
```

Comment:
@sns-seb It would be nice to add code sample here

Reply:
@sns-seb And/or may be we could add call to `addDeprecatedRuleKey` to first chunk of docs in this file

Reply:
incoming :)